### DISCUSSION TITLE
What Reader-Hardware do you recommend?

### DISCUSSION THREAD

**Discussion by henfri** [omersiar/esp-rfid#541] [General]
Hello,

I have tried out three different readers now that also have Fingerprint and Keypad support.
All three are really a pain. A combination of:

- Fingerprint reading not reliable
- Teaching Fingerprints/Keys not reliable/difficult "user interface" via key-codes and no proper feedback "where you are" in the programming
- Reporting cards not reliable
- Only one Pin programable
- Fingers are reported as incrementing UIDs starting with 1 (so not secure)

Is there any hardware you can recommend?

Best regards,
Hendrik

**Comment by matjack1** [comment]
@henfri I've not worked with fingerprint sensors, so if you need that, I cannot recommend any. For a RFID one I'm using a couple Wiegand sensors that are working fine. Normally looking for "Wiegand keypad" gets you in a good spot!
